nautobot/ipam/models.py

```diff
@@ -656,6 +671,14 @@ def reparent_subnets(self):
 
         return query.update(parent=self)
 
+    def reparent_ips(self):
+        """Determine the list of child IPAddresses and set the parent to self."""
+        query = IPAddress.objects.select_for_update().filter(
+            parent_id=self.parent_id,
+        )
```

Comment:
This is kind of feeling like it's not complete enough, but I'm not sure what else needs to be considered right now seeing as the namespace and prefix length are coming from the parent, and the parent should still have enough space to contain them.